packages/react-bindings/src/styles/getStyles.ts

```diff
@@ -74,12 +74,19 @@ const getStyles = (options: GetStylesOptions): GetStylesResult => {
     : undefined
 
   // Resolve styles using resolved variables, merge results, allow props.styles to override
-  const mergedStyles: ComponentSlotStylesPrepared = mergeComponentStyles(
-    theme.componentStyles[displayName],
-    props.design && withDebugId({ root: props.design }, 'props.design'),
-    props.styles && withDebugId({ root: props.styles } as ComponentSlotStylesInput, 'props.styles'),
-    animationStyles && withDebugId({ root: animationStyles }, 'props.animation'),
-  )
+  let mergedStyles: ComponentSlotStylesPrepared = theme.componentStyles[displayName] || {
+    root: () => ({}),
+  }
+
+  if (!_.isNil(props.design) || !_.isNil(props.animation) || !_.isNil(props.styles)) {
```

Comment:
If there are inline overrides, then these will be merged on top of the theme, and it would be more expensive to use them..

Reply:
isn't `!(_isNil(props.design) && ...)` faster?

Reply:
...probably not

Reply:
```suggestion
const hasInlineOverrides = !_.isNil(props.design) || !_.isNil(props.animation) || !_.isNil(props.styles))
```

Let's extract it to something meaningful